lib/vnet/setup.go

```diff
@@ -0,0 +1,130 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package vnet
+
+import (
+	"context"
+	"log/slog"
+	"os"
+
+	"github.com/gravitational/trace"
+	"golang.org/x/sync/errgroup"
+	"golang.zx2c4.com/wireguard/tun"
+)
+
+// Run is a blocking call to create and start Teleport VNet.
+func Run(ctx context.Context) error {
```

Comment:
I slightly disagree because this is responsible for calling all the other setup code, and then starting the vnet, and i'd rather leave it to avoid conflicts with later PRs as well